packages/console/src/pages/AuthStatusChecker/index.tsx

```diff
@@ -0,0 +1,392 @@
+    localStorage.removeItem(testKey);
+
+    if (testValue !== 'test') {
+      return { accessible: false, error: 'localStorage read/write test failed' };
+    }
+
+    return { accessible: true };
+  } catch (error) {
+    return {
+      accessible: false,
+      error: error instanceof Error ? error.message : 'Unknown localStorage error',
+    };
+  }
+};
+
+/**
+ * Request storage access using the Storage Access API
+ */
+const requestStorageAccess = async (): Promise<{
+  granted: boolean;
+  hasAccess: boolean;
+  error?: string;
+}> => {
+  try {
+    // Check if Storage Access API is available
+    if (!('requestStorageAccess' in document) || !('hasStorageAccess' in document)) {
+      return { granted: false, hasAccess: false, error: 'Storage Access API not available' };
+    }
+
+    // Check if we already have storage access
+    const hasAccess = await document.hasStorageAccess();
+    if (hasAccess) {
+      return { granted: true, hasAccess: true };
+    }
+
+    // Request storage access
+    await document.requestStorageAccess();
+    const newHasAccess = await document.hasStorageAccess();
+
+    return { granted: newHasAccess, hasAccess: newHasAccess };
+  } catch (error) {
+    return {
+      granted: false,
+      hasAccess: false,
+      error: error instanceof Error ? error.message : 'Storage access request failed',
+    };
+  }
+};
+
+/**
+ * Cross-domain authentication status checker page
+ * This page is designed to be embedded as an iframe to check login status
+ * across same-origin subdomains and localhost for development.
+ */
+function AuthStatusChecker() {
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
+  const [debugLogs, setDebugLogs] = useState<DebugLogEntry[]>([]);
+  const [isDebugMode, setIsDebugMode] = useState(false);
+  const [currentToken, setCurrentToken] = useState<string | undefined>(undefined);
+
+  const urlParams = new URLSearchParams(window.location.search);
+  const debugParam = urlParams.get('debug');
+  const debugFlag = localStorage.getItem(DEBUG_MODE_KEY);
+
+  // Check if we're in debug mode (query parameter or localStorage flag)
+  useEffect(() => {
+    if (yes(debugParam) || yes(debugFlag)) {
+      setIsDebugMode(true);
+    }
+  }, [debugParam, debugFlag]);
+
+  // Add debug log entry
+  const addDebugLog = useCallback(
+    (type: DebugLogEntry['type'], message: string, data?: unknown) => {
+      // Skip logging if debug mode is not enabled.
+      if (!isDebugMode) {
+        return;
+      }
+
+      const logEntry: DebugLogEntry = {
+        timestamp: new Date().toISOString(),
+        type,
+        message,
+        data,
+      };
+
+      console.log(`[AuthStatusChecker] ${type.toUpperCase()}: ${message}`, data);
+      setDebugLogs((previous) => [...previous, logEntry]);
+    },
+    [isDebugMode]
+  );
+
+  // Monitor token changes
+  useEffect(() => {
+    const token = localStorage.getItem(ADMIN_CONSOLE_ID_TOKEN_KEY);
```

Comment:
Since this localStorage key is tied to the SDK's implementation, should we source it from the SDK or a shared constant (import from SDK lib) to make it more robust against future changes?

Reply:
use `useLogto()` hook, `getIdToken()` and `isAuthenticated` method.